spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/src/main/java/org/springframework/cloud/function/adapter/gcloud/GcfSpringBootHttpRequestHandler2.java

```diff
@@ -69,4 +69,11 @@ public void service(HttpRequest httpRequest, HttpResponse httpResponse) throws E
 		httpResponse.getWriter().write(gson.toJson(result));
 		httpResponse.getWriter().close();
 	}
+
+	@Override
+	protected String doResolveName(Object targetContext) {
+		HttpRequest context = (HttpRequest) targetContext;
+		return context.getFirstHeader("spring.function").orElse("");
```

Comment:
This is actually problematic because the `targetContext` is cached from the first request.
See #10.